quill-core/src/main/scala/io/getquill/ast/AstOps.scala

```diff
@@ -0,0 +1,81 @@
+package io.getquill.ast
+
+import io.getquill.norm.BetaReduction
+
+object Implicits {
+  implicit class AstOpsExt(body: Ast) {
```

Comment:
Can we avoid these implicits and use the objects directly? I'm afraid people will see the method available and use it even though it doesn't work for all ast nodes. Also, we could restrict the `object`s to support only `BinaryOperation`

Reply:
I'm suggesting removing only the implicits and calling the objects in this file directly

Reply:
I can certainly remove `reduce`, that makes sense, especially because BetaReduction is a noun.

What about the `+||+` etc.. methods though? Then it would turn into:
```scala
apply(`+||+`((IsNotNullCheck(ast), `+&&+`((IsNotNullCheck(ast), reduction)))): Ast)
```
That's still pretty confusing to me so I'd rather leave them. Wdyt?

Reply:
okay, let's keep them 👍

Reply:
Cool

Reply:
I can't make `+||+` and `+&&+` limited to `BinaryOperation` because of this:
```scala
      case OptionExists(ast, alias, body) =>
        val reduction = BetaReduction(body, alias -> ast)
        apply((IsNotNullCheck(ast) +&&+ reduction): Ast)
```
Since the AST is untyped, there's no way to know that the AST inside OptionExists must yield a `BinaryOperation` instance.